internal/characters/kinich/cons.go

```diff
@@ -55,58 +59,55 @@ func (c *char) c2ResShredCB(a combat.AttackCB) {
 	})
 }
 
-func (c *char) C2Snapshot(ai combat.AttackInfo, dmgBonus int) combat.Snapshot {
-	s := c.Snapshot(&ai)
-	s.Stats[attributes.DmgP] = float64(dmgBonus)
-	if dmgBonus > 0 {
-		c.Core.Log.NewEvent("Kinich C2 Damage Bonus", glog.LogCharacterEvent, c.Index).
-			Write("bonus", dmgBonus).
-			Write("final", s.Stats[attributes.DmgP])
+func (c *char) c2Bonus(ai *combat.AttackInfo) (combat.Snapshot, float64) {
+	s := c.Snapshot(ai)
+	if c.Base.Cons < 2 {
+		return s, 3.0
 	}
-	return s
+	if c.c2AoeIncreased {
+		return s, 3.0
+	}
+	c.c2AoeIncreased = true
+	s.Stats[attributes.DmgP] += 1.0
+	c.Core.Log.NewEvent("Kinich C2 Damage Bonus", glog.LogCharacterEvent, c.Index).
+		Write("final", s.Stats[attributes.DmgP])
+	return s, 5.0
 }
 
 func (c *char) c4() {
 	if c.Base.Cons < 4 {
 		return
 	}
-	c.Core.Events.Subscribe(event.OnEnemyHit, func(args ...interface{}) bool {
-		atk := args[1].(*combat.AttackEvent)
-		switch atk.Info.AttackTag {
-		case attacks.AttackTagElementalArt:
-		case attacks.AttackTagElementalArtHold:
-		default:
-			return false
-		}
-
-		if atk.Info.ActorIndex == c.Index {
-			return false
-		}
-
-		active := c.Core.Player.ActiveChar()
-		if active.Index == atk.Info.ActorIndex {
-			return false
-		}
-		if c.StatusIsActive(c4IcdKey) {
-			return false
-		}
-		c.AddStatus(c4IcdKey, 2.8*60, true)
-		c.AddEnergy("kinich-c4", 5)
-
-		return false
-	}, "kinich-c4-energy")
+	if c.StatusIsActive(c4IcdKey) {
+		return
+	}
+	c.AddStatus(c4IcdKey, 2.8*60, true)
+	c.AddEnergy("kinich-c4", 5)
 
 	m := make([]float64, attributes.EndStatType)
 	c.AddAttackMod(character.AttackMod{
 		Base: modifier.NewBase("kinich-c4-dmgp", -1),
 		Amount: func(atk *combat.AttackEvent, t combat.Target) ([]float64, bool) {
-			switch atk.Info.AttackTag {
-			case attacks.AttackTagElementalBurst:
-			default:
+			if atk.Info.AttackTag != attacks.AttackTagElementalBurst {
 				return nil, false
 			}
 			m[attributes.DmgP] = 0.7
 			return m, true
 		},
 	})
 }
+
+func (c *char) c6(ai combat.AttackInfo, s *combat.Snapshot, radius float64, target combat.Target, travel int) {
+	if c.Base.Cons < 6 {
+		return
+	}
+	ai.Abil = c6Abil
+	var next combat.Target = c.Core.Combat.RandomEnemyWithinArea(combat.NewCircleHitOnTarget(target, nil, radius), func(t combat.Enemy) bool {
+		return target.Key() != t.Key()
+	})
+	if next == nil {
```

Comment:
C6 does not bounce to the same target. Need at least 2 target for it to work, so no need for that.

Reply:
https://youtu.be/GpMYeymfi9I?t=79